p2p/host/autorelay/options.go

```diff
@@ -74,18 +72,27 @@ func init() {
 type Option func(*config) error
 
 func WithStaticRelays(static []peer.AddrInfo) Option {
+
+	staticRelayChan := make(chan peer.AddrInfo, len(static))
+
 	return func(c *config) error {
-		if c.setMinCandidates {
-			return errStaticRelaysMinCandidates
-		}
-		if c.peerSource != nil {
-			return errStaticRelaysPeerSource
-		}
-		if len(c.staticRelays) > 0 {
+		if c.staticRelaySource != nil {
 			return errors.New("can't set static relays, static relays already configured")
 		}
 		c.minCandidates = len(static)
-		c.staticRelays = static
+		c.staticRelaySource = func(ctx context.Context, num int) <-chan peer.AddrInfo {
+			c.staticRelayMinInterval = time.Hour
+			rand.Seed(time.Now().UnixNano())
+			var static_candidates = static
+			rand.Shuffle(len(static_candidates), func(i, j int) {
```

Comment:
Why the shuffle? Users may have a preferred order, no?

Reply:
The rationale here was: if asked only for 1 or 2 candidates (`<len(static_candidates)`), and those candidates happened to be always down, then one wouldn't be able to connect to any. Randomizing the process guarantees than you will pull different candidates as longs as `len(static_candidates)>num`. But yes, if order is important, the user can just increase `num` and get more candidates if previous ones haven't succeeded.